src/js/components/AppVersionComponent.jsx

```diff
@@ -10,36 +10,58 @@ var UNSPECIFIED_NODE =
     }
   });
 /*eslint-enable react/display-name */
+
+function invalidateValue(value, suffix) {
+  if (value == null) {
+    return (
+      <UNSPECIFIED_NODE />
+    );
+  } else {
+    return (
+      <dd>{value} {suffix}</dd>
+    );
+  }
+}
+
 var AppVersionComponent = React.createClass({
   displayName: "AppVersionComponent",
 
   propTypes: {
-    app: React.PropTypes.instanceOf(App).isRequired,
-    appVersion: React.PropTypes.oneOfType([
-      React.PropTypes.instanceOf(App).isRequired,
-      React.PropTypes.instanceOf(AppVersion).isRequired
-    ]),
+    // App object
+    appVersion: React.PropTypes.object.isRequired,
     className: React.PropTypes.string,
-    currentVersion: React.PropTypes.bool,
-    onRollback: React.PropTypes.func
+    currentVersion: React.PropTypes.bool
+  },
+
+  handleRollbackToAppVersion: function () {
+    var appVersion = this.props.appVersion;
+    AppsActions.applySettingsOnApp(appVersion.id, appVersion);
   },
 
-  handleSubmit: function (event) {
-    if (_.isFunction(this.props.onRollback)) {
-      event.preventDefault();
-      this.props.onRollback(this.props.appVersion, event);
+  getApplyButton: function () {
+    var applyButton = null;
+
+    if (!this.props.currentVersion) {
+      applyButton = (
+        <div className="text-right">
+          <button type="submit"
+              className="btn btn-sm btn-default"
+              onClick={this.handleRollbackToAppVersion}>
+            Apply these settings
+          </button>
+        </div>
+      );
     }
+
+    return applyButton;
```

Comment:
I don't like them, if they are mixed with markup. Looks messy. :)
In mathematical algorithms I like them too.